pkg/resource/deploy/step_generator.go

```diff
@@ -1355,15 +1355,11 @@ func (sg *stepGenerator) generateStepsFromDiff(
 					return nil, fmt.Errorf("could not load provider for resource %v: %w", old.URN, err)
 				}
 
-				var deleteStep Step
-				if old.PendingReplacement {
-					deleteStep = NewRemovePendingReplaceStep(sg.deployment, old)
-				} else {
-					deleteStep = NewDeleteReplacementStep(sg.deployment, sg.deletes, old, true)
+				if !old.PendingReplacement {
+					steps = append(steps, NewDeleteReplacementStep(sg.deployment, sg.deletes, old, true))
```

Comment:
I'd be tempted to add a comment here explaining that the pending replacement flag will get cleaned up when the replacement completes/the create executes, and that doing nothing is deliberate in this case?